collectors/metrics/cmd/metrics-collector/main.go

```diff
@@ -203,12 +196,17 @@ func main() {
 		opt.LogLevel,
 		"Log filtering level. e.g info, debug, warn, error")
 
-	// deprecated opt
-	cmd.Flags().StringVar(
-		&opt.Identifier,
-		"id",
-		opt.Identifier,
-		"The unique identifier for metrics sent with this client.")
+	cmd.Flags().BoolVar(
```

Comment:
What is the purpose for exposing these flags?

Reply:
e2e tests in the future. I already have one like so, but to add it here, would need some go.mod changes which we can tackle in later PRs.

```go
package e2e_test

import (
	"fmt"
	"os"
	"strconv"
	"testing"
	"time"

	"github.com/efficientgo/e2e"
	e2edb "github.com/efficientgo/e2e/db"
	e2einteractive "github.com/efficientgo/e2e/interactive"
	e2emon "github.com/efficientgo/e2e/monitoring"
	e2eobs "github.com/efficientgo/e2e/observable"
	"github.com/thanos-io/thanos/test/e2e/e2ethanos"

	"github.com/efficientgo/core/backoff"

	"github.com/efficientgo/core/testutil"
)

func TestMetricsCollectorE2E(t *testing.T) {
	e, err := e2e.NewDockerEnvironment("metrics-col")
	testutil.Ok(t, err)
	t.Cleanup(e.Close)

	m, err := e2emon.Start(e)
	testutil.Ok(t, err)

	prom := e2edb.NewPrometheus(e, "prom2")
	receive1 := e2ethanos.NewReceiveBuilder(e, "receiver-1").WithImage("quay.io/thanos/thanos:v0.35.1").WithIngestionEnabled().Init()

	testutil.Ok(t, e2e.StartAndWaitReady(prom, receive1))

	collector := NewMetricsCollector(e, "collector-1", MetricsCollectorOptions{
		From:     prom.InternalEndpoint("http"),
		To:       receive1.InternalEndpoint("remote-write"),
		Matchers: []string{`{__name__="up"}`, `{__name__="go_gc_duration_seconds"}`, `{__name__="go_goroutines"}`, `{__name__="go_memstats_alloc_bytes"}`, `{__name__="go_memstats_heap_alloc_bytes"}`, `{__name__="go_memstats_heap_idle_bytes"}`, `{__name__="go_memstats_heap_inuse_bytes"}`, `{__name__="go_memstats_heap_objects"}`, `{__name__="go_memstats_heap_released_bytes"}`, `{__name__="go_memstats_heap_sys_bytes"}`, `{__name__="go_memstats_last_gc_time_seconds"}`, `{__name__="go_memstats_lookups_total"}`, `{__name__="go_memstats_mallocs_total"}`, `{__name__="go_memstats_sys_bytes"}`, `{__name__="go_threads"}`, `{__name__="process_cpu_seconds_total"}`, `{__name__="process_open_fds"}`, `{__name__="process_resident_memory_bytes"}`, `{__name__="process_start_time_seconds"}`, `{__name__="process_virtual_memory_bytes"}`},
		// Matchers: []string{`{__name__="up"}`},
		LogLevel: "info",
		Interval: "2s",
		Workers:  4,
	})

	testutil.Ok(t, e2e.StartAndWaitReady(collector))

	time.Sleep(10 * time.Second)

	testutil.Ok(t, receive1.WaitSumMetricsWithOptions(e2emon.GreaterOrEqual(10), []string{"prometheus_tsdb_head_series"}, e2emon.WaitMissingMetrics()))

	query1 := e2edb.NewThanosQuerier(
		e,
		"query1",
		[]string{
			receive1.InternalEndpoint("grpc"),
		},
		e2edb.WithImage("quay.io/thanos/thanos:v0.35.1"),
	)
	testutil.Ok(t, e2e.StartAndWaitReady(query1))
	testutil.Ok(t, query1.WaitSumMetricsWithOptions(e2emon.Equals(1), []string{"thanos_store_nodes_grpc_connections"}, e2emon.WaitMissingMetrics()))
	testutil.Ok(t, e2einteractive.OpenInBrowser(fmt.Sprintf("http://%s", prom.Endpoint("http"))))
	testutil.Ok(t, e2einteractive.OpenInBrowser(fmt.Sprintf("http://%s", query1.Endpoint("http"))))
	testutil.Ok(t, m.OpenUserInterfaceInBrowser())
	testutil.Ok(t, e2einteractive.RunUntilEndpointHit())

}

type MetricsCollectorOptions struct {
	From     string
	To       string
	Matchers []string
	LogLevel string
	Interval string
	Workers  int
}

func NewMetricsCollector(e e2e.Environment, name string, o MetricsCollectorOptions) *e2eobs.Observable {
	f := e.Runnable(name).WithPorts(map[string]int{"metrics": 9002}).Future()

	args := e2e.BuildArgs(map[string]string{
		"--from":                     "http://" + o.From,
		"--to-upload":                "http://" + o.To + "/api/v1/receive",
		"--log-level":                o.LogLevel,
		"--interval":                 o.Interval,
		"--disable-status-reporting": "true",
		"--disable-hypershift":       "true",
		"--worker-number":            strconv.Itoa(o.Workers),
	})

	for _, matcher := range o.Matchers {
		args = append(args, "--match="+matcher)
	}

	return e2eobs.AsObservable(f.Init(wrapWithDefaults(e2e.StartOptions{
		Image:   "example.com/metrics-collector:v0.0.1",
		Command: e2e.NewCommandWithoutEntrypoint("metrics-collector", args...),
	})), "metrics")
}

var defaultBackoffConfig = backoff.Config{
	Min:        300 * time.Millisecond,
	Max:        600 * time.Millisecond,
	MaxRetries: 50,
}

func wrapWithDefaults(opt e2e.StartOptions) e2e.StartOptions {
	if opt.User == "" {
		opt.User = strconv.Itoa(os.Getuid())
	}
	if opt.WaitReadyBackoff == nil {
		opt.WaitReadyBackoff = &defaultBackoffConfig
	}
	return opt
}

```